src/Interpreters/TreeRewriter.cpp

```diff
@@ -976,7 +976,7 @@ TreeRewriterResultPtr TreeRewriter::analyzeSelect(
     result.required_source_columns_before_expanding_alias_columns = result.required_source_columns.getNames();
 
     /// rewrite filters for select query, must go after getArrayJoinedColumns
-    if (settings.optimize_respect_aliases && result.metadata_snapshot)
+    if (settings.optimize_respect_aliases && result.metadata_snapshot && !select_options.hasShardInfo())
```

Comment:
So we cannot expand aliases when it's not an initial query. We have a better check for this:

https://github.com/ClickHouse/ClickHouse/blob/7ee2dc04a1f67963e7e32db9873031d325a0c2bb/src/Interpreters/ExpressionAnalyzer.cpp#L252